### PR TITLE
Add RemainingDisplayWidget component

### PR DESCRIPTION
**Pull request by ArmandoGraterol**
Added new RemainingDisplayWidget component to CoinMachine in order to display time and tokens remaining

**New stuff** ✨

* Added `question-mark-inverted` icon
* Added `invertedIcon` prop to QuestionMarkTooltip component to flip the icon's color
* Added `RemainingDisplayWidget` to CoinMachine
* Added `TokenPriceStatusIcon` component to `RemainingDisplayWidget`

### How to test each case
1. The first case is the one with placeholders so basically just enter the coin machine page
2. The second case is the one where values are added so just add some string like `0:00:00` or `1200/2500` to the `value` prop of the widget. Also, in the case of the tokens remaining footer, in order to see the price status icon, you need to pass a `TokenPriceStatuses` value to the widget with a `footerText`.
3. The third case is the one with red background and you just need to pass an `appearance={{ theme: 'danger' }}` to the widget.
4. The final case is when the values are displayed in red, just change the `false` to `true` on line 64 of the RemainingDisplayWidget.

FireShot Capture 391 - Colony - localhost

FireShot Capture 392 - Colony - localhost

FireShot Capture 393 - Colony - localhost

FireShot Capture 394 - Colony - localhost

FireShot Capture 395 - Colony - localhost

FireShot Capture 396 - Colony - localhost

Resolves DEV-453

### CODE REVIEW

**Review by chinins — COMMENTED** [review]
Have some comments/suggestions (please, see my notes).

File: src/modules/dashboard/components/CoinMachine/RemainingDisplayWidget/RemainingDisplayWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+type Props = {
+  title: string | MessageDescriptor;
+  value: string | number | null;
+  tooltipText: string | MessageDescriptor;
+  placeholderText: string | MessageDescriptor;
+  footerText?: string | MessageDescriptor;
+  appearance?: Appearance;
+  tokenPriceStatus?: TokenPriceStatuses;
+};
```
[the remark this answers is not in the document]
Comment (alicjakujawa) [reply src/modules/dashboard/components/CoinMachine/RemainingDisplayWidget/RemainingDisplayWidget.tsx:27]: Agree with this, logic should be handled in RemainingDisplayWidget based on provided values

**Review by alicjakujawa — CHANGES_REQUESTED** [review]

**Review by chinins — APPROVED** [review]
Good to go!

**Review by alicjakujawa — CHANGES_REQUESTED** [review]
Nice job with moving logic to component itself. Unfortunately we still miss part of logic, component which display "time remaning" its actually counter, you can check time in SaleStateWidget and copy/reuse it

**Comment by alicjakujawa** [comment]
> Nice job with moving logic to component itself. Unfortunately we still miss part of logic, component which display "time remaning" its actually counter, you can check time in SaleStateWidget and copy/reuse it

ah and one more thing, lets be consistent and stick with same format of time displaying like in SalesStateWidget (and motions and disputes)

**Comment by alicjakujawa** [comment]
> > Nice job with moving logic to component itself. Unfortunately we still miss part of logic, component which display "time remaning" its actually counter, you can check time in SaleStateWidget and copy/reuse it
> 
> ah and one more thing, lets be consistent and stick with same format of time displaying like in SalesStateWidget (and motions and disputes)

You can also extract timer logic/styles etc and reuse it in both components

**Comment by ArmandoGraterol** [comment]
> Nice job with moving logic to component itself. Unfortunately we still miss part of logic, component which display "time remaning" its actually counter, you can check time in SaleStateWidget and copy/reuse it

My understanding was that this kind of logic wouldn't be included on this issue, but oh well if the logic is already done in another place and I just need to reuse it then I'll do it

**Review by alicjakujawa — CHANGES_REQUESTED** [review]
Screenshot 2021-07-28 at 08 42 58
Same issue with grey background

**Review by alicjakujawa — APPROVED** [review]
Looks good, please only update font-weight to normal for value (as in designs we dont have bold) Pre-approving

**Comment by alicjakujawa** [comment]
> Looks good, please only update font-weight to normal for value (as in designs we dont have bold) Pre-approving

aha, and rebase and test haha

**Comment by ArmandoGraterol** [comment]
> Looks good, please only update font-weight to normal for value (as in designs we don't have bold) Pre-approving

Hmmm, in the design the values look like they are on `700`.

I found that in Figma **only** the font-weight of `N/A`  is supposed to be "400", but the font-weight of the other values is "700". Also, if I use "400" then they wouldn't look like they appear in the design.

So basically, I think Figma is smoking crack because there's no way the "N/A" is using `400` and it looks like that. It must be using `700` like the rest of the values.

**Pull request merged**